src/crypto/tls_mbedtls_alt.c

```diff
@@ -1730,7 +1730,7 @@ static int tls_mbedtls_set_params(struct tls_conf *tls_conf, const struct tls_co
     int ret = mbedtls_ssl_config_defaults(
         &tls_conf->conf, tls_ctx_global.tls_conf ? MBEDTLS_SSL_IS_SERVER : MBEDTLS_SSL_IS_CLIENT,
         MBEDTLS_SSL_TRANSPORT_STREAM,
-        (tls_conf->flags & TLS_CONN_SUITEB) ? MBEDTLS_SSL_PRESET_SUITEB : MBEDTLS_SSL_PRESET_DEFAULT);
+        MBEDTLS_SSL_PRESET_DEFAULT);
```

Comment:
By RSA-3K, you are referring to https://github.com/zephyrproject-rtos/hostap/blob/main/src/crypto/tls_mbedtls_alt.c#L1679. AFAIK, RSA isn't part of SUITE-B (even the 3K), at least as per https://en.wikipedia.org/wiki/NSA_Suite_B_Cryptography# and the MbedTLS behaviour seems to be correct. And event if it is this looks like a MbedTLS bug, the RSA-3K should be added to SUITE-B list.

Can you point to some literature about RSA-3K being part of SUITE-B?

Reply:
hi @krish2718 ,

Yes, RSA isn't part of EmbedTLS Suite-B. 

At first EmbedTLS Suite B profile code has RSA included.
Then it appears that this bug report triggered the RSA removal: 
https://github.com/Mbed-TLS/mbedtls/issues/8221.

Some literature for your reference:
•	Initially as per https://datatracker.ietf.org/doc/html/rfc5430 and https://datatracker.ietf.org/doc/html/rfc6460, only ECC is supported
•	DH and RSA was set as “legacy” to be discontinued by October 1st 2015
•	This was revised by CNSSP No. 15 from Oct 2016:
https://www.cnss.gov/CNSS/openDoc.cfm?a=nQUTpKgoOqqkPFoNb%2BoF3w%3D%3D&b=467C780853881A895A4162494D0AA5D3DE499CE5938855DCE183B428C4F217E570B30FBF762CA332098990819F0D808A

According to https://datatracker.ietf.org/doc/html/rfc5430 :
[3](https://datatracker.ietf.org/doc/html/rfc5430#section-3).  Suite B Requirements
   The Fact Sheet on Suite B Cryptography requires that key
   establishment and authentication algorithms be based on Elliptic
   Curve Cryptography, and that the encryption algorithm be [AES](https://datatracker.ietf.org/doc/html/rfc5430#ref-AES) [AES].
   Suite B defines two security levels, of 128 and 192 bits.

There is no RSA in suite B. So the removal of RSA to MbedTLS is correct.

Then for WPA3 enterprise suiteb192 rsa3k:
We will use MBEDTLS_SSL_PRESET_DEFAULT to make conf->sig_algs include RSA algorithm, 
instead of use MBEDTLS_SSL_PRESET_SUITEB which profile is without RSA algorithm.

Thanks.
Li Long

Reply:
> • This was revised by CNSSP No. 15 from Oct 2016:
https://www.cnss.gov/CNSS/openDoc.cfm?a=nQUTpKgoOqqkPFoNb%2BoF3w%3D%3D&b=467C780853881A895A4162494D0AA5D3DE499CE5938855DCE183B428C4F217E570B30FBF762CA332098990819F0D808A

Yes, even Wiki link above says `In 2018, NSA replaced Suite B with the Commercial National Security Algorithm Suite (CNSA).[2]`.

So, technically `suiteb192 rsa3k` is wrong, it should be `CNSA-RSA3K`, I still think a proper way would be to introduce a new preset called CNSA and add RSA3K in MbedTLS https://github.com/Mbed-TLS/mbedtls/blob/195e1647b24d231521675c0aab6d4715bb579be5/library/ssl_tls.c#L5978

Reply:
https://github.com/Mbed-TLS/mbedtls/issues/4602 3 year old issue :) and first task

> Define presets for CNSA, similar to the existing presets for NSA Suite B.

This is exactly what we need. WDYT?

Reply:
I can try and attempt a PR to MbedTLS

Reply:
Here's my attempt: https://github.com/Mbed-TLS/mbedtls/pull/9460

Reply:
> Here's my attempt: [Mbed-TLS/mbedtls#9460](https://github.com/Mbed-TLS/mbedtls/pull/9460)

hi @krish2718 , 
please review the new change to run with your attempt which add CNSA presets, it could success do STA connect with WPA3 enterprise suiteb192 RSA 3k case as in WPA3 spec 19.5.2 step 4.

Reply:
Thanks for confirming. Can you please cherry-pick the MbedTLS and update
https://github.com/zephyrproject-rtos/zephyr/pull/76826

Reply:
This is akin to `fromlist`

Reply:
> Thanks for confirming. Can you please cherry-pick the MbedTLS and update [zephyrproject-rtos/zephyr#76826](https://github.com/zephyrproject-rtos/zephyr/pull/76826)

hi @krish2718 , 
Since your are the author of CNSA preset change in MbedTLS, could you please cherry-pick the change.
Then I could update https://github.com/zephyrproject-rtos/zephyr/pull/76826 to include both PRs (mbedtls PR and hostap PR).
Thanks.

Reply:
Sorry it took so long, its done now https://github.com/zephyrproject-rtos/zephyr/pull/78785